specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,737 @@
+* Show details of snapshot
+##### Request
+```json
+  “id”: “string”
+```  
+##### Response parameters
+```json
+  “snapshotId”: “string”
+
+  “protocols” : “array”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 8.   POST /v1beta/{tenantId}/file/snapshot
+* Creates a snapshot from a share.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+
+  “shareId” : “string“
+```
+  ##### Response parameters
+```json  
+  “Id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : “array”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 9.	PUT /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Updates a share snapshot.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+```  
+##### Response parameters
+```json
+  “snapshotId”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : “array”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 10.	DELETE /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Deletes a share
+##### Request
+```json
+  “snapshotId” : “sting”
+```
+##### 11.	POST /v1beta/{tenantId}/file/accesses/
+* Add access control for file share.
+##### Request
+```json
+    “sharetId” : “sting”
+
+    “type” : “string”    // user/ip
+
+    “accessCapability” : read/write/execute
+
+    “accessTo” : “user_list”
+```
+##### 12.	PUT /v1beta/{tenantId}/file/accesses/{accessId}
+* Add access control for file share.
+##### Request
+```json
+   “accessId” : “string”
+```
+## Implementation of file share APIs for Profiles consolidated with block storage
+Note : Highlighted with background color texts are specific to file share and those are only supported in file share. Rest all parameters are belongs to block storage
+
+##### 1.	POST /v1beta/{tenantId}/profiles
+* Create profiles
+
+##### Request
+```json
+body*
+   [
+     ""name"         : "string",
+     "storage type" : "string",
+     "description"  : "string",
+     "provisioningProperties" : {
+                "datastore""  : {
+
+                   ""characterCodeSet"        : "string",
+                    "maxFileNameLengthBytes"  : "int64",
+                    "storageAccessCapability" : "string",
+                    "recoveryTimeObjective"   : "int64",
+                    "provisioningPolicy"      : "string",
+                    "isSpaceEfficient"        : "bool"
+                },
+                "ioconnectivity" : {
+                  "accessProtocol" : "string",
```

Comment:
Yes it means NFS/CIFS, based on the create file share request, there should be profile selected. Profile selects dock that contacts the particular driver so profile should have accessprotocol information to direct the request to the supported protocol drivers to create file share

Reply:
Thank you for your reply.
I understand that profile should have accessprotocol information.

I have an another question.
The create file share request parameter also has protocols parameter.
Could you tell me relationship between protocols parameter and accessprotocol parameter of profile?
What happens if both protocols parameter and accessprotocol parameter of profile are specified at the same time, which one has priority?

Reply:
@hirokikmr Thanks for your review!  That's a very good observation.  I don't think we need to have a separate "protocols" parameter in Create file share request since it is already in Profiles.

@Shruthi-1MN and @PravinRanjan10, I think the Protocols in Create File share can be removed.  We can just use the accessProtocol in Profiles instead.  Let's make this "accessProtocol" an array ([]string).

Reply:
@xing-yang Thank you for your reply. I agree with your suggestion.

Reply:
@xing-yang @hirokikmr, i have removed the "protocols" parameter from POST API of fileshare. And also corrected "accessProtocol" an array ([]string).

Reply:
AccessProtocol is a string, they should choose only one accessprotocol not multiple.